lib/datasets.go

```diff
@@ -1009,16 +1009,24 @@ func (m *DatasetMethods) Validate(p *ValidateParams, res *ValidateResponse) erro
 		return qrierr.New(ErrBadArgs, "please provide a dataset name, or a supply the --body and --schema or --structure flags")
 	}
 
-	ref, err := repo.ParseDatasetRef(p.Ref)
-	if err != nil && err != repo.ErrEmptyRef {
-		return err
-	}
-	err = repo.CanonicalizeDatasetRef(m.inst.repo, &ref)
-	if err != nil && err != repo.ErrEmptyRef {
-		if err == repo.ErrNotFound {
-			return fmt.Errorf("cannot find dataset: %s", ref)
+	fsiPath := ""
+	var err error
+	ref := dsref.Ref{}
+
+	// if there is both a bodyfilename and a schema/structure
+	// we don't need to resolve any references
+	if !(p.BodyFilename != "" && schemaFlagType != "") {
+		// TODO (ramfox): we need consts in `dsref` for "local", "network", "p2p"
```

Comment:
Again, not sure this is the time, but keep seeing "local" "network" and "p2p" in a bunch of places. Might be nice to have some codified consts that we can rely on.

Reply:
I think this would be great to have, I would even go further and say that the resolution source should be it's own type, which is either some known instance like `LocalSource` or `P2PSource`, or something wrapped custom address like `RemoveSource("http://my-remote-url/").